app/firebase.client.ts

```diff
@@ -0,0 +1,18 @@
+import { initializeApp } from "firebase/app";
+import { connectAuthEmulator, getAuth } from "firebase/auth";
+import { FirebaseClientConfig } from "./types";
+
+const AUTH_EMULATOR_URL = "http://localhost:9099";
+
+export function initFirebaseClient(config: FirebaseClientConfig) {
```

Comment:
Because we display a GitHub popup, we have to handle some authentication on the client. That's taken care of by the `firebase` package, and it's the reason why I checked in some credentials.